lib/commons/dom/has-lang-text.js

```diff
@@ -0,0 +1,24 @@
+import { hasChildTextNodes } from './has-content-virtual';
+import isVisualContent from './is-visual-content';
+import isVisible from './is-visible';
+
+/**
+ * Check that a node has text, or an accessible name which language is defined by the 
+ * nearest ancestor's lang attribute.
+ * @param {VirtualNode} virtualNode 
+ * @return boolean
+ */
+export default function hasLangText(virtualNode) {
+  if (typeof virtualNode.children === 'undefined' || hasChildTextNodes(virtualNode)) {
+    return true;
+  }
+  if (virtualNode.props.nodeType === 1 && isVisualContent(virtualNode)) {
+    // See: https://github.com/dequelabs/axe-core/issues/3281
+    return !!axe.commons.text.accessibleTextVirtual(virtualNode);
```

Comment:
this call here is going to be slow

Reply:
Possibly, although it'll be rare. You generally don't see people put lang attributes on images with aria-labelledby or whatever.